src/storage-preview/azext_storage_preview/track2_util.py

```diff
@@ -21,3 +21,28 @@ def make_file_url(client, directory_name, file_name, sas_token=None):
         url += '?' + sas_token
 
     return url
+
+
+def list_generator(pages, num_results):
+    result = []
+
+    # get first page items
+    page = list(next(pages))
+    result += page
+
+    while True:
+        if not pages.continuation_token:
+            break
+
+        # handle num results
+        if num_results is not None:
+            if num_results == len(result):
+                if pages.continuation_token:
+                    next_marker = {"nextMarker": pages.continuation_token}
+                    result.append(next_marker)
```

Comment:
I noticed that in main repo, we add `next_marker` to result or warning message according to `--show-next-marker`. Why we don't need such parameter and just make adding to result as default behavior?

Reply:
as warning message is legacy method, to not break customer, we still keep it. Adding next marker in result will give customer better experience to get the marker value instead of parsing stderr.